packages/peregrine/src/Router/resolveUnknownRoute.js

```diff
@@ -77,10 +77,28 @@ function remotelyResolveRoute(opts) {
             });
         }
     } else {
-        return fetchRoute(opts);
+        return checkRoute(opts);
     }
 }
 
+/**
+ * @description Calls the GraphQL API for results from the urlResolver query
+ * @param {{ route: string, apiBase: string}} opts
+ * @returns {Promise<resolve>}
+ */
+async function checkRoute(opts)
+    {
+      let resolve = await fetchRoute(opts);
+      if (resolve === null) {
+        if (opts.route.endsWith('.html')) {
+          opts.route = opts.route.replace('.html', '');
+          resolve = await fetchRoute(opts)
```

Comment:
Let's say the user clicks on an external ink to `venia.com/accessories.html`. If that catalog page no longer exists, this function will attempt to look up the entity at `accessories`. If that's a CMS page, it will display the CMS page without changing the URL in the browser location. Is this what we want?

I think there may be a better place to remove the URL key than this.

Reply:
Catalog categories have .html  aswell, so unless a cms page is called /checkout then thats the only time i think this wont work.

Reply:
What if there used to be an `/accessories.html`, but that is no longer a category, and instead there is a CMS page called `/accessories`? That's where I'm worried about a bug. This may need to just redirect instead of silently removing the `.html` from the active route. Am I missing something?